yogstation/code/game/gamemodes/battle_royale/battleroyale.dm

```diff
@@ -0,0 +1,215 @@
+		var/list/turfs = list()
+		for(var/turf/open/TT in get_area_turfs(AA))
+			if(!is_blocked_turf(TT))
+				turfs += TT
+		if(turfs.len)
+			var/turf/turfy = pick(turfs)
+			new /obj/structure/closet/crate/battleroyale(turfy)
+	return TRUE
+
+/datum/game_mode/fortnite/post_setup() //now add a place for them to spawn :)
+	GLOB.enter_allowed = FALSE
+	message_admins("Battle Royale Mode has disabled late-joining. If you re-enable it you will break everything.")
+	for(var/i = 1 to queued.len)
+		var/datum/mind/virgin = queued[i]
+		SEND_SOUND(virgin.current, 'yogstation/sound/effects/battleroyale/battlebus.ogg')
+		virgin.add_antag_datum(antag_datum_type)
+		if(!GLOB.thebattlebus) //Ruhoh.
+			virgin.current.forceMove(pick(GLOB.start_landmarks_list))
+			message_admins("There is no battle bus! Attempting to spawn players at random.")
+			continue
+		virgin.current.forceMove(GLOB.thebattlebus)
+		virgin.current.add_trait(TRAIT_XRAY_VISION) //so they can see where theyre dropping
+		virgin.current.update_sight()
+		to_chat(virgin.current, "<font_color='red'><b> You are now in the battle bus! Click it to exit </b></font>")
+		GLOB.battleroyale_players += virgin.current
+	addtimer(CALLBACK(src, .proc/check_win), 300)
+	addtimer(CALLBACK(src, .proc/shrinkborders), 10)
+	return ..()
+
+/datum/game_mode/fortnite/check_win()
+	. = ..()
+	var/list/royalers = list()
+	var/area/hallway/secondary/A = locate(/area/hallway/secondary) in GLOB.sortedAreas
+	var/turf/T = pick(get_area_turfs(A)) //Work out the station's Z based on arrivals. Don't want people running off now!
+	var/station_z = T.z
+	if(GLOB.player_list.len <= 1) //It's a localhost testing
+		return
+	for(var/mob/living/player in GLOB.battleroyale_players)
+		if(player.stat == DEAD)
+			GLOB.battleroyale_players -= player
+		if(!player.client)
+			GLOB.battleroyale_players -= player
+			continue //No AFKS allowed!!!
+		if(player.onCentCom())
+			GLOB.battleroyale_players -= player
+			to_chat(player, "You left the station Z-level! You have been disqualified from battle royale.")
+			continue
+		else if(player.onSyndieBase())
+			GLOB.battleroyale_players -= player
+			to_chat(player, "You left the station Z-level! You have been disqualified from battle royale.")
+			continue
+		if(player.z != station_z)
+			GLOB.battleroyale_players -= player
+			to_chat(player, "You left the station Z-level! You have been disqualified from battle royale.")
+			continue
+		royalers += player
+
+	if(!royalers.len)
+		SSticker.mode.check_finished(TRUE)
+		SSticker.force_ending = 1
+		to_chat(world, "<span_class='ratvar'>L. Nobody wins!</span>")
+		SEND_SOUND(world, 'yogstation/sound/effects/battleroyale/L.ogg')
+		return
+	if(royalers.len == 1) //We have a wiener!
+		SSticker.mode.check_finished(TRUE)
+		SSticker.force_ending = 1
+		var/mob/living/dub = pick(royalers)
+		to_chat(world, "<img src='https://cdn.discordapp.com/attachments/351367327184584704/539903688857092106/victoryroyale.png'>")
```

Comment:
❓ ❔ ❓

Reply:
Victory royale.png

Reply:
Does Discord actually permanently store all images?

Reply:
if the message is not deleted, or the channel isn't deleted, or the server isn't deleted, then yes

Reply:
What is it with people necroing year-old PRs